ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/future/FutureUtil.scala

```diff
@@ -0,0 +1,14 @@
+package com.daml.ledger.api.testtool.infrastructure.future
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object FutureUtil {
+
+  def sequential[T, R](elements: Seq[T])(f: T => Future[R])(implicit
+      ec: ExecutionContext
+  ): Future[Seq[R]] =
+    elements.foldLeft(Future.successful(Seq.empty[R])) { case (results, newElement) =>
+      results.flatMap(resultsSoFar => f(newElement).map(resultsSoFar :+ _))
```

Comment:
but as Future is starting eagerly, will it really be sequential here? 

How about akka stream instead?
import akka.stream.scaladsl._
Source(elements).mapAsync(1) { case element => f(element) }.runWith(Sink.seq)

Reply:
I wouldn't use akka streams unless we need it. 
It's a simple left fold, so the future is not started until the `flatMap` is actually run, which is done only after any previous future has finished.

Reply:
ah yes, my fault, you are right. in this case I would not use streams of course